packages/firebase_ml_vision/lib/src/barcode_detector.dart

```diff
@@ -4,18 +4,452 @@
 
 part of firebase_ml_vision;
 
+/// Detector for performing optical character recognition(OCR) on an input image.
+///
+/// A barcode detector is created via getVisionBarcodeDetector() in [FirebaseVision]:
+///
+/// ```dart
+/// BarcodeDetector barcodeDetector = FirebaseVision.instance.barcodeDetector();
+/// ```
 class BarcodeDetector extends FirebaseVisionDetector {
-  BarcodeDetector._(BarcodeDetectorOptions options);
+  BarcodeDetector._();
 
+  /// Closes the barcode detector and release its model resources.
   @override
   Future<void> close() async {
-    // TODO: implement close
+    return FirebaseVision.channel.invokeMethod('BarcodeDetector#close');
   }
 
+  /// Detects barcode in the input image.
+  ///
+  /// The OCR is performed asynchronously.
   @override
   Future<void> detectInImage(FirebaseVisionImage visionImage) async {
-    // TODO: implement detectInImage
+    final List<dynamic> reply = await FirebaseVision.channel.invokeMethod(
+      'BarcodeDetector#detectInImage',
+      visionImage.imageFile.path,
+    );
+
+    final List<VisionBarcode> barcodes = <VisionBarcode>[];
+    reply.forEach((dynamic barcode) {
+      barcodes.add(new VisionBarcode._(barcode));
+    });
+
+    return barcodes;
   }
 }
 
-class BarcodeDetectorOptions {}
+class VisionBarcode {
+  final Rectangle<int> boundingBox;
+  final String rawValue;
+  final String displayValue;
+  final VisionBarcodeFormat format;
+  final List<Point<int>> cornerPoints;
+  final VisionBarcodeValueType valueType;
+  final VisionBarcodeEmail email;
+  final VisionBarcodePhone phone;
+  final VisionBarcodeSMS sms;
+  final VisionBarcodeURLBookmark url;
+  final VisionBarcodeWiFi wifi;
+  final VisionBarcodeGeoPoint geoPoint;
+  final VisionBarcodeContactInfo contactInfo;
+  final VisionBarcodeCalendarEvent calendarEvent;
+  final VisionBarcodeDriverLicense driverLicense;
+
+  VisionBarcode._(Map<dynamic, dynamic> _data)
+      : boundingBox = Rectangle<int>(
+          _data['left'],
+          _data['top'],
+          _data['width'],
+          _data['height'],
+        ),
+        rawValue = _data['raw_value'] != null ? _data['raw_value'] : null,
```

Comment:
If you're aim is to set rawValue to `null` if `_data['raw_value']` equals null. I believe you can just use the line `rawValue = _data['raw_value'],`

Reply:
You are right!